packages/mongodb-mql-model/src/docs/md/bson-type/bson-type.md

```diff
@@ -0,0 +1,151 @@
+# MQL BSON Type
+-----------
+
+## Abstract
+
+This specification documents the different kinds of BSON types and how they are related to the
+original source code of an [MQL Query](../mql-query/mql-query.md). This document aims to provide
+information about the behaviour of dialects and linters on the computation of the original
+expression BSON type.
+
+## META
+
+The keywords "MUST", "MUST NOT", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY"
+and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Specification
+
+[BSON](https://bsonspec.org/spec.html) is a binary format that is used to communicate between the
+MongoDB Client (through a driver) and a MongoDB Cluster. MQL BSON (from now on we will just say BSON) 
+is a superset of the original BSON types.
```

Comment:
It's a subset? `{ $akjsdhkj: {} }` is valid BSON but not valid MQL

Reply:
It's a valid BsonObject right, even if it's not a valid query.

Reply:
I mean, I just repeated what you said, let me rephrase: a BsonType is how we represent parts of an MQL query, but it's not necessarily a valid query per se. For example, a BsonString is not a valid query, but it's still a valid type. This specific document models the types, independently of their use in a query.

Reply:
When I mean superset I mean because there are semantics that BSON does not have, like BsonAnyOf btw.

Reply:
Alright, I'd maybe mention that as an example here then :)

Reply:
Done, thanks for pointing this out!